.github/CONTRIBUTING.md

```diff
@@ -15,45 +27,50 @@ git clone git@github.com:lightspeed/flame.git
 To create a brand new component, use the following command that will create a base skeleton ready for development:
 
 ```sh
-yarn generate Component
+yarn generate
 ```
 
-Where `Component` is the name of the component you want to create. This will create the base component structure and story for development/documentation in `packages/flame-{component-name}`, which are considered [add-ons / experimental](https://github.com/lightspeed/flame#add-ons-and-experimental-components) until being deemed ready to go into the [mainline package](https://github.com/lightspeed/flame/tree/master/packages/flame).
+You will be given a choice of "Mainline" or "Add-on" component, then have to choose a name for the component.
+
+#### Mainline component
+
+Mainline components will be created under `packages/flame/src/{ComponentName}`, where all other components are located. This will create the base component structure ready for development, including README for docs, story for Storybook development and visual regression test, tests and TypeScript component file.
+
+#### Add-on component
+
+Add-on component will be created under the `packages/flame-{component-name}`. It should be used for add-on/experimental components that are not ready to go into the [mainline package](https://github.com/lightspeed/flame/tree/master/packages/flame) or are considered as "extras" and don't fit our component library.
+
+Add-on components are a completely separate package with their own version and have `@lightspeed/flame` and all its required dependencies as `peerDependencies`. They include the same base component structure as a mainline component, but also have a `tsconfig` to output types as well as their own LICENSE and CHANGELOG files.
```

Comment:
I don't think we need to be too granular in the component structure (re: CHANGELOG and LICENSE) if you word this section as follows:

```suggestion
Add-on components are a completely separate package which are versioned independently, licensed independently, and have `@lightspeed/flame` and all its required dependencies as `peerDependencies`. They include the same base component structure as a mainline component.
```

Independent versioning and licensing implies that the package has its own LICENSE and CHANGELOG.